src/utils/builder.ts

```diff
@@ -268,6 +268,32 @@ export function setStateCriteriaSource(
   };
 }
 
+export function setStateAction(pathway: Pathway, key: string, action: Action[]): Pathway {
```

Comment:
When a guidance state is created it initialized the action to be an empty list. I suggest making this method `addStateAction` to create a new action. Then use the other set action methods to set the specific attributes of the action

Reply:
isn't this already accomplished inside the action node?  What benefit do we get from swapping stuff around like this?